### PR TITLE
Fog routers infra changes

### PR DESCRIPTION
**Pull request by joekottke**
* Helm charts, Dockerfiles, and GitHub Actions updates. Also uses new cluster issuer in Development.
* Actual work completed in #3191 and #3157 (which this PR supercedes). This is a cleaned up and PR to master branch.

### Motivation

Changes for deploying fog routers (ledger and view).  Only deployment and dockerfile changes.

### Future Work

Naming consistency updates, and some cleanup.

### CODE REVIEW

File: .internal-ci/docker/Dockerfile.fogview [hunk 1]
```diff
@@ -12,6 +12,7 @@ FROM ${REPO_ORG}/runtime-base:${BASE_TAG}
 ARG RUST_BIN_PATH=target/release
 COPY ${RUST_BIN_PATH}/libview-enclave.signed.so /usr/bin/
 COPY ${RUST_BIN_PATH}/fog_view_server /usr/bin/
```
Comment (jgreat) [inline .internal-ci/docker/Dockerfile.fogview:14]: In fog_ledger `fog_ledger_router` replaces the `fog_ledger_server`  should this be the same here? Where fog_view_server should be removed.

File: .internal-ci/helm/fog-services/templates/fog-view-router-grpc-ingress.yaml
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-view-router-grpc-ingress.yaml:1]: I don't think we want to rename this file, this should just be named `fog-view-grpc-ingress.yaml`

File: .internal-ci/helm/fog-services/templates/fog-view-router-http-ingress.yaml
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-view-router-http-ingress.yaml:1]: I don't think we want to rename this file, this should just be named fog-view-http-ingress.yaml

File: .internal-ci/helm/fog-services/templates/fog-view-store-svc.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    app: fog-view
+    {{- include "fogServices.labels" . | nindent 4 }}
+spec:
+  type: ClusterIP
+  clusterIP: None
+  selector:
+    app: fog-view-store
+    {{- include "fogServices.selectorLabels" . | nindent 4 }}
+  ports:
+    - name: view
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-view-store-svc.yaml:16]: Follow patterns from fog-view-service.yaml: 

```
- name: view-grpc
  port: 3225
  targetPort: view-grpc
- name: mgmt-http
    port: 8000
    targetPort: mgmt-http
  - name: view-http
    port: 8225
    targetPort: view-http
```

File: .internal-ci/helm/fog-services/templates/fog-view-store-svc.yaml
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/fog-view-store-svc.yaml:1]: Follow existing file name patterns: `fog-view-store-service.yaml`

File: .internal-ci/helm/fog-services/templates/supervisord-fog-ledger-configmap.yaml [hunk 3]
```diff
@@ -10,15 +10,15 @@ data:
     [program:fog-ledger]
     priority=100
     environment=MC_SENTRY_DSN="%(ENV_FOG_LEDGER_SENTRY_DSN)s"
-    command=/usr/bin/ledger_server
+    command=/usr/bin/key_image_store
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/supervisord-fog-ledger-configmap.yaml:13]: This is an upstream question.  Why is this not called `fog-ledger-server` like `fog-view-server`

Should we change all the referencing names  to `key-image`?

File: .internal-ci/helm/fog-services/templates/supervisord-fog-router-configmap.yaml
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/supervisord-fog-router-configmap.yaml:1]: Should be named `supervisord-fog-view-router-configmap.yaml`

File: .internal-ci/helm/fog-services/templates/supervisord-fog-router-configmap.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "fogServices.fullname" . }}-supervisord-fog-router
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/templates/supervisord-fog-router-configmap.yaml:4]: supervisord-fog-view-router

File: .internal-ci/helm/fog-services/values.yaml [hunk 5]
```diff
@@ -87,7 +135,7 @@ fogReport:
       POSTGRES_MAX_CONNECTIONS: '3'
 
 fogView:
-  replicaCount: 4
+  replicaCount: 3
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/values.yaml:138]: Is this now the `router` or the `store` replica count

File: .internal-ci/helm/fog-services/values.yaml [hunk 6]
```diff
@@ -138,8 +186,40 @@ fogView:
...
+
+  tolerations:
+  - key: sgx
+    operator: Equal
+    value: 'true'
+    effect: NoSchedule
+
 fogLedger:
-  replicaCount: 2
+  replicaCount: 3
```
Comment (jgreat) [inline .internal-ci/helm/fog-services/values.yaml:222]: Is this now the router or the store replica count

**Review by jgreat — CHANGES_REQUESTED** [review]
We could use a quick round of work to make resource names and descriptors consistent.  

Doing this kind of clean up work after the fact is much more error prone, the unexpected ingress/service routing on the last fog-view deployment is an example.

 I suggest we clearly mark all resources with `router` or `store`.
 
I'd also suggest we clean up the rust binary names here: 
- Use `-` not `_` in binary names
- Rename `key_image_store` to `fog-ledger-store` OR rename `key_image_store` to `key-image-store` and rename all associated resources to from `fog-ledger` to `key-image`

File: .github/workflows/mobilecoin-workflow-dev-deploy.yaml [hunk 7]
```diff
@@ -86,8 +86,8 @@ jobs:
               intel.com/sgx: 2000
           persistence:
             enabled: false
-          client:
```
[the remark this answers is not in the document]
Comment (jgreat) [reply .github/workflows/mobilecoin-workflow-dev-deploy.yaml:89]: Just a note, no fix needed here.

File: .internal-ci/helm/fog-services/templates/supervisord-fog-ledger-configmap.yaml (hunk 3 above)
Comment (awygle) [reply .internal-ci/helm/fog-services/templates/supervisord-fog-ledger-configmap.yaml:13]: The Ledger service serves multiple APIs, one of which is the Key Image API. The Key Image API is the only one that has been converted to a router-type format. Hence why this is Key Image Store and not Ledger Store - it only backs the Key Image API.

Fog View on the other hand only serves one API. We could talk about `fog-view-server` vs `fog-view-store`, but that's the primary difference.

File: .internal-ci/docker/Dockerfile.fogview (hunk 1 above)
Comment (awygle) [reply .internal-ci/docker/Dockerfile.fogview:14]: Without taking a position on what the names should be, these are the names that are currently being used.

`fog_view_server` - the Fog View store server
`fog_view_router` - the Fog View router server
`key_image_store` - the Fog Ledger Key Image store server
`fog_ledger_router` - the Fog Ledger router server

File: .internal-ci/helm/consensus-node/templates/client-grpc-attest-ingress.yaml [hunk 8]
```diff
@@ -5,7 +5,7 @@ kind: Ingress
 metadata:
   name: {{ include "consensusNode.fullname" . }}-client-grpc-attest
   annotations:
-    {{- if .Values.node.ingress.clientAttest.rateLimits.enabled }}
```
[the remark this answers is not in the document]
Comment (joekottke) [reply .internal-ci/helm/consensus-node/templates/client-grpc-attest-ingress.yaml:8]: Reverted to previous usage.

File: .internal-ci/helm/consensus-node/templates/client-http-attest-ingress.yaml [hunk 9]
```diff
@@ -5,7 +5,7 @@ kind: Ingress
 metadata:
   name: {{ include "consensusNode.fullname" . }}-client-http-attest
   annotations:
-    {{- if .Values.node.ingress.clientAttest.rateLimits.enabled }}
```
[the remark this answers is not in the document]
Comment (joekottke) [reply .internal-ci/helm/consensus-node/templates/client-http-attest-ingress.yaml:8]: Reverted to previous usage.

File: .internal-ci/helm/fog-services/templates/fog-view-router-http-ingress.yaml
Comment (joekottke) [reply .internal-ci/helm/fog-services/templates/fog-view-router-http-ingress.yaml:1]: Will handle in follow-up PR.

File: .internal-ci/helm/fog-services/templates/fog-view-store-svc.yaml
Comment (joekottke) [reply .internal-ci/helm/fog-services/templates/fog-view-store-svc.yaml:1]: Will handle in follow-up PR.

File: .internal-ci/helm/fog-services/templates/fog-view-store-svc.yaml (hunk 2 above)
Comment (joekottke) [reply .internal-ci/helm/fog-services/templates/fog-view-store-svc.yaml:16]: Will handle in follow-up PR.

File: .internal-ci/helm/fog-services/templates/supervisord-fog-ledger-router.yaml
[the remark this answers is not in the document]
Comment (joekottke) [reply .internal-ci/helm/fog-services/templates/supervisord-fog-ledger-router.yaml:1]: Will handle in follow-up PR.

File: .internal-ci/helm/fog-services/templates/supervisord-fog-router-configmap.yaml
Comment (joekottke) [reply .internal-ci/helm/fog-services/templates/supervisord-fog-router-configmap.yaml:1]: Will handle in follow-up PR.

File: .internal-ci/helm/fog-services/templates/supervisord-fog-router-configmap.yaml (hunk 4 above)
Comment (joekottke) [reply .internal-ci/helm/fog-services/templates/supervisord-fog-router-configmap.yaml:4]: Will handle in follow-up PR.

File: .internal-ci/helm/fog-services/values.yaml (hunk 6 above)
Comment (joekottke) [reply .internal-ci/helm/fog-services/values.yaml:222]: Will address in follow-up PR.  Needs spelunking.

File: .internal-ci/helm/fog-services/templates/fog-view-router-grpc-ingress.yaml
Comment (joekottke) [reply .internal-ci/helm/fog-services/templates/fog-view-router-grpc-ingress.yaml:1]: Agreed.  Will follow-up in clean-up PR.

File: .internal-ci/helm/consensus-node/templates/client-grpc-attest-ingress.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (joekottke) [reply .internal-ci/helm/consensus-node/templates/client-grpc-attest-ingress.yaml]: This has been fixed and removed from this PR.

File: .internal-ci/helm/consensus-node/templates/client-http-attest-ingress.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (joekottke) [reply .internal-ci/helm/consensus-node/templates/client-http-attest-ingress.yaml]: This has been fixed and removed from this PR.

**Review by awygle — APPROVED** [review]

**Review by jgreat — APPROVED** [review]

**Pull request merged**